include/ginkgo/config.hpp.in

```diff
@@ -13,6 +13,16 @@
 #define GKO_VERSION_STR @Ginkgo_VERSION_MAJOR@, @Ginkgo_VERSION_MINOR@, @Ginkgo_VERSION_PATCH@
 // clang-format on
 
+
+// clang-format off
+#cmakedefine01 GINKGO_VERSION_TAG_DEPRECATED
+#if GINKGO_VERSION_TAG_DEPRECATED
+#pragma message "The version tag " GKO_VERSION_TAG " is deprecated and will stop receiving updates after 2025. " \
+ "Please use the main branch for the latest release, or the develop branch for the latest development updates."
```

Comment:
In MSVC, you need to use `#pragma message WARN( ...)` from lower_trs.hpp

Reply:
Is `WARN` a macro from us, because it doesn't seem to be part of msvc: https://learn.microsoft.com/en-us/cpp/preprocessor/message?view=msvc-160?

Reply:
I am also checking, but do not get anything specifying WARN when we have that in the beginning.

Reply:
Maybe set the tag as master temporary to check whether it report the message on MSVC

Reply:
That's a good suggestion, thanks!

Reply:
You can see here that it does complain quite a lot: https://gitlab.com/ginkgo-project/ginkgo-public-ci/-/jobs/8557710509